source/content/guides/drupal-9-migration/06-build-tools-to-d9-build-tools.md

```diff
@@ -0,0 +1,281 @@
+
+1. After the build has finished without error, you will see a new environment in the Dashboard under **Multidev** named in reference your pull request.
+
+   ```bash{promptUser: user}
+   terminus drush $SITE.$ENV pm-enable upgrade_status --yes
+   ```
+
+1. Create a one-time login to your site with the following command:
+
+   ```bash{promptUser: user}
+   terminus drush $SITE.$ENV uli admin
+   ```
+
+1. Log into the site as admin and take a look under **Reports** at **Upgrade Status**. Any modules Upgrade Status says are incompatible will need to be updated in the next few steps. Take note of the versions **Upgrade Status** recommends. If your module is incompatible it will need to be removed from the Composer file.
+
+## Custom Module Code
+
+Custom module code is outside the scope of this document. See [drupal.org](https://www.drupal.org/docs/creating-custom-modules) for getting your custom code updated with the new version numbers and any code deprecations.
+
+## Use Composer to Update Drupal Core
+
+1. Temporarily add write access to protected files and directories:
+
+   ```bash{promptUser: user}
+   chmod 777 web/sites/default
+   find web/sites/default -name "*settings.php" -exec chmod 777 {} \;
+   find web/sites/default -name "*services.yml" -exec chmod 777 {} \;
+   ```
+
+1. Use Composer to remove `config_installer` and add new requirements:
+
+   ```bash{outputLines: 3-7,9-11}
+   composer remove drupal/config_installer --no-update
+   composer require drupal/core-recommended:^9 \
+      drupal/core-composer-scaffold:^9 \
+      drupal/core-project-message:^9 \
+      drush/drush:^10 \
+      -W --no-update
+
+   composer require phpunit/phpunit:^9 \
+      behat/behat:^3 \
+      drupal/drupal-extension:^4 \
+      --no-update -W --dev
+   ```
+
+1. If you have `core-dev` installed:
+
+   ```bash{outputLines: 2}
+   composer require drupal/core-dev:^9 \
+      --dev -W --no-update
+   ```
+
+1. If `UPGRADE STATUS` displays obsolete modules, update them using the `--no-update -W` switch to instruct Composer to check for all dependencies together rather than for each module. Replace `OBSOLETE-MODULE-NAME` in this example with the module to update:
+
+   ```bash{outputLines: 2}
+   composer require drupal/OBSOLETE-MODULE-NAME:^9 \
+      --dev -W --no-update
+   ```
+
+   Repeat this step for each obsolete module in the project.
+
+1. When you're done updating modules, run `composer update`:
+
+```bash{promptUser: user}
+composer update -W --optimize-autoloader --prefer-dist
+```
+
+   If this command returns an error, check the output for any incompatible module or themes and check the **Upgrade Status** in the integration environment.
+
+1. Edit `composer.json` and remove `--no-dev` from the `scripts` section:
```

Comment:
is the intent to add `--no-dev` in this step? The step says `remove`, but L259 under `To:` has it replaced into the section